abel/rbasex.py

```diff
@@ -0,0 +1,751 @@
+    IM : m × n numpy array
+        the image to be transformed
+    origin : tuple of int or str
+        image origin, explicit in the (row, column) format, or as a location
+        string
+    rmax : int or string
+        largest radius to include in the transform
+    order : int
+        highest angular order present in the data, ≥ 0
+    odd : bool
+        include odd angular orders (enabled automatically if **order** is odd)
+    weights : m × n numpy array, optional
+        weighting factors for each pixel. The array shape must match the image
+        shape. Parts of the image can be excluded from analysis by assigning
+        zero weights to their pixels.
+    direction : str: ``'forward'`` or ``'inverse'``
+        type of Abel transform to be performed
+    reg : None or str or tuple (str, float), optional
+        regularization to use for inverse Abel transform. ``None`` means no
+        regularization, a string selects a non-parameterized regularization
+        method, and parameterized methods are selected by a tuple (`method`,
+        `strength`). Available methods are:
+
+        ``'L2'``:
+            Tikhonov :math:`L_2` regularization with `strength` as the square
+            of the Tikhonov factor. This is the same as “Tikhonov
+            regularization” used in BASEX, with almost identical effects on the
+            radial distributions.
+        ``'diff'``:
+            Tikhonov regularization with the difference operator (approximation
+            of the derivative) multiplied by the square root of `strength` as
+            the Tikhonov matrix. This tends to produce less blurring, but more
+            negative overshoots than ``'L2'``.
+        ``'SVD'``:
+            truncated SVD (singular value decomposition) with
+            N = `strength` × **rmax** largest singular values removed for each
+            angular order. This mimics the approach proposed in pBasex.
+        ``'pos'``:
+            non-parameterized method, finds the best (in the least-squares
+            sense) solution with non-negative :math:`\cos^n\theta \sin^m\theta`
+            terms (see :meth:`~abel.tools.vmi.Distributions.Results.cossin`).
+            For **order** = 0, 1, and 2 (with **odd** = `False`) this is
+            equivalent to :math:`I(r, \theta) \geqslant 0`; for higher orders
+            this assumption is stronger than :math:`I \geqslant 0` and
+            corresponds to no interference between different multiphoton
+            channels. Not implemented for odd orders > 1.
+
+            Notice that this method is nonlinear, which also means that it is
+            considerably slower than the linear methods and the transform
+            operator cannot be cached.
+
+        In all cases, `strength` = 0 provides no regularization. For the
+        Tikhonov methods, `strength` ~ 100 is a reasonable value for megapixel
+        images. For truncated SVD, `strength` must be < 1; `strength` ~ 0.1 is
```

Comment:
For the SVD method, is the effect of the `strength` independent of image size? For the Tikhonov methods, should we try to find a way to scale the strength of the regularization so that the effect is independent of image size? It might be something simply like simply multiplying the strength by the image size...  It would be nice if we could just recommend a strength and it would more-or-less work for all images.

Reply:
I don't actually know. For SVD, I use a fraction instead of an absolute number, since it, in theory, should be less size-dependent. But it still has somewhat different effects for different image sizes. The “L2” method essentially folows the BASEX approach, and its `strength` actually gives effects on velocity distributions very similar to that of BASEX `reg`. So:
1. This question of the optimal regularization strength needs deeper research. I have seen some attempts in the literature, but no definitive answer.
2. So if we want to give more specific recommendations, they perhaps deserve a separate documentation section (with references, examples and so on).
3. I actually have some examples in `transform_methods/rbasex-math.html#examples`, but if you take `rbasex-regRMSE.py` and try it with different images and even different random samples of the _same_ image, you'll see that the optimal parameters will be different. (Hence there is a big “Warning” preceding the examples.)
3. If we want to introduce some automatic scaling to reduce the size dependence, it probably should be done for BASEX as well, for consistency.

Reply:
Added a hyperlink to the end of that paragraph.